src/main/java/com/aws/greengrass/authorization/AuthorizationHandler.java

```diff
@@ -419,12 +421,23 @@ private void addPermission(String destination,
                             Permission.builder().principal(principal).operation(operation).resource(null).build());
                 } else {
                     for (String resource : resources) {
-                        authModule.addPermission(destination,
-                                Permission.builder()
-                                        .principal(principal)
-                                        .operation(operation)
-                                        .resource(resource)
-                                        .build());
+                        try {
+                            authModule.addPermission(destination,
+                                    Permission.builder()
+                                            .principal(principal)
+                                            .operation(operation)
+                                            .resource(resource)
+                                            .build());
+                        } catch (AuthorizationException e) {
+                            logger.atError("load-authorization-config-add-resource-error").setCause(e)
+                                    .kv("policyId", policyId)
+                                    .kv("component", principal)
+                                    .kv("operation", operation)
+                                    .kv("IPC service", destination)
+                                    .kv("resource", resource)
+                                            .log("Error while adding permission for component {} "
+                                                    + "to IPC Service {}", principal, destination);
```

Comment:
What causes the error here? Can we explain it to the customer, including information about what step to take to resolve the error?

Reply:
The error reason will follow this message. The reason comes from the 3 cases that we talk about below